### PR TITLE
ref: felt and build

### LINKED ISSUE #115: Refactor the felt.js and the felt/hex management in order for it to be cleaner

### PR DESCRIPTION
**Pull request by fricoben**
Should fix the build & close #115

### CODE REVIEW

File: utils/stringService.ts [hunk 1]
```diff
@@ -75,3 +66,9 @@ export function isStarkDomain(domain: string) {
     domain
   );
 }
+
+export function numberToString(element: number | undefined): string {
+  if (!element) return "";
```
Comment (Th0rgal) [inline utils/stringService.ts]: Shouldn't you check that element is undefined? Here you might return "" for the integer 0.

File: utils/feltService.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+export function stringToHex(str: string): string {
+  if (!str) return "";
+  const buffer = Buffer.from(str, "utf-8");
+
+  return "0x" + buffer.toString("hex");
+}
+
+export function toHex(element: string | number | undefined): string {
+  if (!element) return "";
```
Comment (Th0rgal) [inline utils/feltService.ts]: Same issue here, number = 0 should be hex "0x0", not ""

File: package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     "react-dom": "18.2.0",
     "react-icons": "^4.4.0",
     "react-loader-spinner": "^5.2.0",
-    "starknet": "^4.4.2",
+    "starknet": "^4.17.1",
```
Comment (Th0rgal) [inline package.json:29]: why should we follow the dev branch of starknetjs?

**Review by Th0rgal — CHANGES_REQUESTED** [review]
Please tell me more about the none check at the beginning of converting functions

File: package.json (hunk 3 above)
Comment (fricoben) [reply package.json:29]: It's not the dev branch is it ?

Comment (fricoben) [reply package.json:29]: It's just the last version. 

Dev branch is @next

**Review by Th0rgal — APPROVED** [review]
good job, let's get this merged

**Pull request merged**